subsys/bluetooth/services/fast_pair/fp_storage/fp_storage_ak.c

```diff
@@ -224,6 +296,186 @@ static int validate_ak_order(void)
 	return 0;
 }
 
+static int bond_name_gen(char *name, uint8_t index)
+{
+	int n;
+
+	n = snprintf(name, SETTINGS_BOND_NAME_MAX_SIZE, "%s%u", SETTINGS_BOND_FULL_PREFIX, index);
+	__ASSERT_NO_MSG(n < SETTINGS_BOND_NAME_MAX_SIZE);
+	if (n < 0) {
+		return n;
+	}
+
+	return 0;
+}
+
+static int fp_settings_bond_delete(size_t bond_idx)
+{
+	int err;
+	char name[SETTINGS_BOND_NAME_MAX_SIZE];
+
+	err = bond_name_gen(name, bond_idx);
+	if (err) {
+		return err;
+	}
+
+	err = settings_delete(name);
+	if (err) {
+		return err;
+	}
+
+	return 0;
+}
+
+static int bond_delete(size_t bond_idx)
+{
+	int err;
+
+	err = fp_settings_bond_delete(bond_idx);
+	if (err) {
+		return err;
+	}
+
+	fp_bonds[bond_idx] = (struct fp_bond){0};
+
+	return 0;
+}
+
+static int bond_remove(size_t bond_idx)
```

Comment:
`ble_bond_stack_unpair` ?

Reply:
Maybe `bond_remove_completely`. This function not only unpairs the bond but also deletes the FP bond from settings if it's not bonded.